examples/pytorch/logging/pytorch_log_model.ipynb

```diff
@@ -0,0 +1,460 @@
+    "y_test_tensor = torch.tensor(y_test, dtype=torch.long)\n",
+    "\n",
+    "# Create datasets and dataloaders\n",
+    "train_dataset = TensorDataset(X_train_tensor, y_train_tensor)\n",
+    "train_loader = DataLoader(dataset=train_dataset, batch_size=16)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define your pytorch model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "51760571-74ad-4053-9191-b82fb061c3f5",
+     "showTitle": true,
+     "title": "Define your pytorch model"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Define a simple neural network model\n",
+    "class SimpleNN(nn.Module):\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "        self.fc1 = nn.Linear(4, 10)\n",
+    "        self.fc2 = nn.Linear(10, 3)\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        x = torch.relu(self.fc1(x))\n",
+    "        x = self.fc2(x)\n",
+    "        return x\n",
+    "\n",
+    "\n",
+    "model = SimpleNN()\n",
+    "criterion = nn.CrossEntropyLoss()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define a simple accuracy metric"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "b52198cc-39af-4a6f-b814-0f3d70eb6e7e",
+     "showTitle": true,
+     "title": "Define a simple accuracy metric"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Function to calculate accuracy\n",
+    "def calculate_accuracy(y_pred, y_true):\n",
+    "    _, predicted = torch.max(y_pred, 1)\n",
+    "    correct = (predicted == y_true).sum().item()\n",
+    "    accuracy = correct / y_true.size(0)\n",
+    "    return accuracy"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Define the model signature\n",
+    "[Link to the `mlflow.models.infer_signature` docs](https://mlflow.org/docs/latest/python_api/mlflow.models.html#mlflow.models.infer_signature)\n",
+    "\n",
+    "A model signature defines valid input and output schema and is used to validate input and outputs at inference time. `mlflow.models.infer_signature` helps with this and creates a `ModelSignature` object that is passed into the `mlflow.pytorch.log_model`.\n",
+    "\n",
+    "Since pytorch model's usually operate on tensors, we need to convert both the input and output into a type compatible with `mlflow.models.infer_signature`. Commonly, this means converting them into a `numpy.ndarray` or a dictionary of arrays (if the output is multiple tensors).\n",
```

Comment:
Should it be `a dictionary of numpy arrays`? In the new schema, if it's a dictionary of lists, we infer it as ColSpec with Array type instead of TensorSpec.

Reply:
+1 let's not make any references to the pre 2.10.0 behavior for signature inference :)